store/tikv/region_cache.go

```diff
@@ -115,6 +115,31 @@ func (r *RegionStore) follower(seed uint32) int32 {
 	return r.workTiKVIdx
 }
 
+// return next leader or follower store's index
+func (r *RegionStore) peer(seed uint32) int32 {
+	l := uint32(len(r.stores))
+	if l <= 1 {
+		return r.workTiKVIdx
+	}
+
+	candidates := make([]int32, 0, len(r.stores))
+	for i := 0; i < len(r.stores); i++ {
+		if r.stores[i].storeType != kv.TiKV {
+			continue
+		}
+		if r.storeFails[i] != atomic.LoadUint32(&r.stores[i].fail) {
+			continue
+		}
+		candidates = append(candidates, int32(i))
+	}
+
+	if len(candidates) == 0 {
+		return r.workTiKVIdx
+	} else {
+		return int32(seed) % int32(len(candidates))
```

Comment:
`stores` value are `[leader, tiflash, tiflash, follower, follower]`
`seed` be 2, `len(candidates)` be 3  and this method will return idx = `2` then `r.getStorePeer` will return a tiflash node in `AnyStorePeer`

Reply:
or stores value are `[leader, follower(fail), follower(fail), follower(ok), follower(ok)]` have same problem

Reply:
Fixed. PTAL.